lib/opentelemetry/wrapper.ts

```diff
@@ -0,0 +1,44 @@
+import { getCurrentMeasurementScope, isRoute06User } from "@/app/(auth)/lib";
+import { createLogger } from "./log";
+import type { ExternalServiceName, RequestCountSchema } from "./types";
+
+export async function withMeasurement<T>(
+	operation: () => Promise<T>,
+	externalServiceName: ExternalServiceName,
+) {
+	const logger = createLogger(externalServiceName);
+	const startTime = performance.now();
+
+	try {
+		const result = await operation();
+		const duration = performance.now() - startTime;
+
+		Promise.all([getCurrentMeasurementScope(), isRoute06User()])
+			.then(([measurementScope, isR06User]) => {
+				const metrics: RequestCountSchema = {
+					externalServiceName,
+					requestCount: 1,
+					duration,
+					measurementScope,
+					isR06User,
+				};
+
+				logger.info(metrics, `[${externalServiceName}] response obtained`);
+			})
+			.catch((error) => {
+				logger.error(error, "failed to get user info for logging");
+			});
+
+		return result;
+	} catch (error) {
+		if (error instanceof Error) {
+			logger.error(error, `[${externalServiceName}] operation failed`);
+		} else {
+			logger.error(
+				new Error("Unknown error occurred"),
+				`[${externalServiceName}] operation failed`,
+			);
+		}
+		throw error;
```

Comment:
@Rindrics 

In the case of errors occurring during OpenTelemetry instrumentation, wouldn't it be better not to throw an error?  
In my opinion, errors related to observability should not interrupt business logic.  
Instead of throwing an error, I believe it's sufficient to send the error to a bug tracking system like Sentry.

Reply:
yes, you're right!

Instrumentation MUST not interfere our business logic.

I'll correct the error case.

Reply:
I fixed the logging wrapper by https://github.com/giselles-ai/giselle/pull/140/commits/54379563a4c8a8620944d7bc9bb03eb6169aa2a6, but remained this final error throwing as it was because this error is thrown by the business-logic function (passed as `operation` to the logging wrapper). @shige 

On the other hand, I corrected other instrumentation-related errors not to interfere business logic and to be just sent to Sentry